egs2/TEMPLATE/asr1/pyscripts/utils/extract_xvectors.py

```diff
@@ -81,8 +81,37 @@ def __init__(self, args, device):
                 )["model"]
             )
             self.model.to(device).eval()
+        elif self.toolkit == "espnet":
+            from espnet2.bin.spk_inference import Speech2Embedding
+
+            # NOTE(jiatong): set default config file as None
+            # assume config is the same path as the model file
+            speech2embedding_kwargs = dict(
+                batch_size=1,
+                dtype="float32",
+                train_config=None,
+                model_file=args.pretrained_model,
+            )
+
+            if args.pretrained_model.endswith("pth"):
+                logging.info(
+                    "the provided model path is end with pth,"
+                    "assume it not a huggingface model"
+                )
+                model_tag = None
+            else:
+                logging.info(
+                    "the provided model path is not end with pth,"
+                    "assume use huggingface model"
+                )
+                model_tag = args.pretrained_model
+
+            self.speech2embedding = Speech2Embedding.from_pretrained(
+                model_tag=model,
+                **speech2embedding_kwargs,
+            )
 
-    def rawnet_extract_embd(self, audio, n_samples=48000, n_segments=10):
+    def _rawnet_extract_embd(self, audio, n_samples=48000, n_segments=10):
```

Comment:
maybe we can remove this now? 
Okay to move it later through a different PR. But our RawNet now outperforms this one (which I uploaded in my personal repo previously)

Reply:
I would suggest keeping it as a legacy api for previous checkpoints

Reply:
We can always suggest to use our new model.